donut/modules/editor/static/script.js

```diff
@@ -0,0 +1,156 @@
+var TITLE_CUTOFF = 20;
+
+// Creates html
+function run() {
+  var text = document.getElementById('source').value,
+      target = document.getElementById('preview'),
+      converter = new showdown.Converter({strikethrough: true}),
+      html = converter.makeHtml(text);
+  target.innerHTML = html;
+}
+
+// Changes the title of a file
+function change_title(oldTitle){
+  var title = document.getElementById('title').value;
+  var valid = /^[0-9a-zA-Z.\/_\- ]*$/.test(title);
+  if (valid)
+  {
+    $.ajax({
+      url: $SCRIPT_ROOT+'/_change_title',
+      type: 'POST',
+      data:{title:title, old_title:oldTitle},
+      success: function(data) {
+        window.alert("Title Change Sucessfully");
+      },
+      error: function(data){
+        window.alert("Please enter a valid title!");
+      }
+    });
+  }
+  else {
+    window.alert("Please enter a valid title!");
+  }
+}
+
+//###########
+
+function selectedString(string1, string2)
+{
+  var ta = document.getElementById('source');
+  if (ta.value.substring(ta.selectionStart, ta.selectionEnd) == "")
+  {
+    insert(string2, false);
+  }
+  else {
+    insert(string1, true);
+  }
+}
+
+// All markdown related functions
+function insert_heading(size){
+  var string = "";
+  for(var i = 0; i < size; i++)
+  {
+    string += "#";
+  }
+  selectedString(string, string + "Heading_title" + string+ "\n");
+}
+
+function insert_italic(){
+  selectedString("*", "*italicText*");
+}
+
+function insert_bold(){
+  selectedString("**", "**boldText**");
+}
+
+function insert_link(){
+  insert("[linkTitle](example.com)", false);
+}
+
+function insert_image(){
+  insert("![Alt text](url/to/image", false);
+}
+
+function insert_ulist(){
+  insert("* list \n", false);
+}
+
+function insert_olist(){
+  insert("1. list \n", false);
+}
+
+
+function insert(string, selected){
+  var ta = document.getElementById('source');
+  if(selected)
+  {
+    var txt2 = ta.value.substring(0, ta.selectionStart)
```

Comment:
You can directly do ta.value = ... in the if else statements, don't need var txt2